### DISCUSSION TITLE
Long-term Retention of Metrics

### DISCUSSION THREAD

**Discussion by luddite516** [NetApp/harvest#619] [Q&A] [answered]
Our organization is a long-time user of the original Netapp Harvest, which used Graphite as its data store.
With Graphite, it was easy to retain metrics for long term (1 year or more), and automatically downsample them to improve performance and efficiency.
I am currently evaluating Harvest with prometheus, and realize that Prometheus does not downsample metrics and is not intended to support long-term storage.
It appears that I would have to introduce Thanos or something similar, in order to get the same functionality we have today with Graphite.  This will make our deployment more complex and expensive.

Does anyone have any thoughts / suggestions on this topic?

**Comment by cgrinds** [comment, marked as answer]
Other tools and links in this area, no particular order:
- https://github.com/timescale/promscale
- https://github.com/cortexproject/cortex supports S3 storage
- https://github.com/VictoriaMetrics/VictoriaMetrics (supports downsampling)
- https://grafana.com/blog/2020/07/16/how-the-cortex-and-thanos-projects-collaborate-to-make-scaling-prometheus-better-for-all/
